libllvmext/src/main/cpp/LLVMExt.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "LLVMExt.h"
+#include "CodeGen.h"
+
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/TargetSelect.h>
+#include <llvm/InitializePasses.h>
+#include <llvm/IR/Module.h>
+#include <llvm/Transforms/IPO/PassManagerBuilder.h>
+#include <llvm/Target/TargetMachine.h>
+#include <llvm/LTO/Config.h>
+#include <llvm/Analysis/TargetLibraryInfo.h>
+#include <llvm/Transforms/IPO.h>
+#include <llvm/Transforms/IPO/AlwaysInliner.h>
+#include <llvm/Support/TargetRegistry.h>
+#include <llvm/Linker/Linker.h>
+#include <llvm/IR/Verifier.h>
+#include <llvm/IR/LegacyPassManager.h>
+#include <llvm/IR/DiagnosticPrinter.h>
+#include <llvm/CodeGen/Passes.h>
+#include <llvm/Support/ToolOutputFile.h>
+#include <llvm/Analysis/TargetTransformInfo.h>
+#include <llvm/IR/IRPrintingPasses.h>
+#include <llvm/MC/SubtargetFeature.h>
+#include <llvm/Support/Host.h>
+
+#include <memory>
+#include <ctime>
+
+using namespace llvm;
+
+class ModuleLinker {
+ public:
+  explicit ModuleLinker(LLVMContext &context)
+      : mergedModule(new Module("merged", context)),
```

Comment:
Did you test that this LLVM launcher doesn't leak native memory allocated by LLVM APIs?